cipher/xor/xor.go

```diff
@@ -2,24 +2,58 @@
 // ref: https://en.wikipedia.org/wiki/XOR_cipher
 package xor
 
+import "unsafe"
+
 // Encrypt encrypts with Xor encryption after converting each character to byte
 // The returned value might not be readable because there is no guarantee
 // which is within the ASCII range
 // If using other type such as string, []int, or some other types,
 // add the statements for converting the type to []byte.
 func Encrypt(key byte, plaintext []byte) []byte {
-	cipherText := []byte{}
-	for _, ch := range plaintext {
-		cipherText = append(cipherText, key^ch)
+	cipherText := make([]byte, len(plaintext))
```

Comment:
Write your implementation in a separate function if it is a different algorithm from the one already implemented.

Reply:
The xor package is to xor each byte in the []byte with the key, and my implementation is designed to do the same.

Reply:
Yes but the previous implementation is also correct. I have no evidence to see why you implementation is better therefore keep the old implemenation and write your implementation in a new function instead of changing the already created (well working) function. Afterwards write benchmarks to test the optimised performances of both implementation that way anyone reading the code can see why your implementation is better/worse than the other implementation.